contracts/StorageFees.cdc

```diff
@@ -0,0 +1,244 @@
+//     StorageFees.storageReservationPath,
+//     target: StorageFees.storageReservationReceiverPath)
+// ```
+pub contract StorageFees {
+    // Emitted when storage capacity refunding is enabled or disabled.
+    pub event RefundingEnabledChanged(_ enabled: Bool)
+    
+    // Emitted when the amount of storage capacity an account has per reserved Flow token changes
+    pub event StorageBytesPerReservedFlowChanged(_ storageBytesPerReservedFlow: UFix64)
+
+    // Emitted when minimum amount of Flow 
+    pub event MinimumStorageReservationChanged(_ minimumStorageReservation: UFix64)
+
+    // Emitted when the minimum amount of Flow tokens that an account needs to have reserved for storage capacity changes.
+    pub event StorageReservationChanged(address: Address, oldStorageReservation: UFix64, oldStorageCapacity: UInt64, newStorageReservation: UFix64, newStorageCapacity: UInt64)
+
+    // Defines the path where each account should have a `StorageReservationReceiver` capability
+    pub let storageReservationReceiverPath: PublicPath
+
+    // Defines the path where each account should have a `StorageReservation` capability
+    pub let storageReservationPath: StoragePath
+
+    // Defines how much storage capacity an account has per reserved Flow token.
+    // definition is written per unit of flow instead of the inverse, so there is no loss of precision calculating storage from flow, but there is loss of precision when calculating flow per storage.
+    pub var storageBytesPerReservedFlow: UFix64
+
+    // Defines the minimum amount of Flow tokens that an account needs to have reserved for storage capacity.
+    // If an account has less then this amount reserved by the end of any transaction it participated in, the transaction will fail.
+    pub var minimumStorageReservation: UFix64
+
+    // Enables or disables the refunding storage capacity.
+    pub var refundingEnabled: Bool
+
+    // An administrator resource that can change the parameters of the StorageFees smart contract.
+    pub resource Administrator {
+        pub fun setRefundingEnabled(_ enabled: Bool) {
+            if StorageFees.refundingEnabled == enabled {
+              return
+            }
+            StorageFees.refundingEnabled = enabled
+            emit RefundingEnabledChanged(enabled)
+        }
+
+        // Changes the amount storage capacity an account has per accounts' reserved storage.
+        pub fun setStorageBytesPerReservedFlow(_ storageBytesPerReservedFlow: UFix64) {
+            if StorageFees.storageBytesPerReservedFlow == storageBytesPerReservedFlow {
+              return
+            }
+            StorageFees.storageBytesPerReservedFlow = storageBytesPerReservedFlow
+            emit StorageBytesPerReservedFlowChanged(storageBytesPerReservedFlow)
+        }
+
+        // Changes the minimum amount of Flow tokens an account has to have reserved.
+        pub fun setMinimumStorageReservation(_ minimumStorageReservation: UFix64) {
```

Comment:
Same comment as above about pre-conditions

Reply:
No. As far as I understand the minimum can be anything.